src/ACEtk/block/details/BaseCrossSectionBlock.hpp

```diff
@@ -0,0 +1,115 @@
+#ifndef NJOY_ACETK_BLOCK_BASECROSSSECTIONBLOCK
+#define NJOY_ACETK_BLOCK_BASECROSSSECTIONBLOCK
+
+// system includes
+
+// other includes
+#include "ACEtk/block/details/Base.hpp"
+
+namespace njoy {
+namespace ACEtk {
+namespace block {
+namespace details {
+
+/**
+ *  @class
+ *  @brief The base class for a LSIG and SIG block with cross section data
+ */
+template < typename Data >
+class BaseCrossSectionBlock : protected details::Base {
+
+  /* fields */
+  unsigned int ntr_; // the number of reactions (excluding elastic)
+  Iterator sig_;     // the begin iterator of the SIG block
+
+  /* auxiliary functions */
+  #include "ACEtk/block/details/BaseCrossSectionBlock/src/generateXSS.hpp"
+  #include "ACEtk/block/details/BaseCrossSectionBlock/src/verifySize.hpp"
+
+protected:
+
+  /* constructor */
+  #include "ACEtk/block/details/BaseCrossSectionBlock/src/ctor.hpp"
+
+  /**
+   *  @brief Return the iterator to the start of the sig block
+   */
+  Iterator sig() const { return this->sig_; }
```

Comment:
[clang-tidy]
This one's interesting. `ctor.hpp` already declared the block after it as public. As a result, the function `Iterator sig()` is not protected. I think this is the general weakness to using the `#include` fragments.

Reply:
This class was refactored later (into BasBlockWIthLocators) and this issue was resolved as well. The ctor file now declares one private constructor while all others are protected. The equivalant to the sig() function is thus also protected.

I do agree that these types of issue can pop up due to the use of include files.